inlong-sort/sort-connectors/iceberg/src/main/java/org/apache/inlong/sort/iceberg/sink/collections/RocksDBDAO.java

```diff
@@ -0,0 +1,523 @@
+     * @param columnFamilyName Column Family name
+     * @param key Key to be deleted
+     */
+    public void delete(String columnFamilyName, String key) {
+        try {
+            getRocksDB().delete(managedHandlesMap.get(columnFamilyName), key.getBytes());
+        } catch (RocksDBException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * Perform a single Delete operation.
+     *
+     * @param columnFamilyName Column Family name
+     * @param key Key to be deleted
+     */
+    public void delete(String columnFamilyName, K key) {
+        try {
+            getRocksDB().delete(managedHandlesMap.get(columnFamilyName), serializeKey(key));
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * Retrieve a value for a given key in a column family.
+     *
+     * @param columnFamilyName Column Family Name
+     * @param key Key to be retrieved
+     */
+    public V get(String columnFamilyName, String key) {
+        Preconditions.checkArgument(!closed);
+        try {
+            byte[] val = getRocksDB().get(managedHandlesMap.get(columnFamilyName), key.getBytes());
+            return val == null ? null : deserializeValue(val);
+        } catch (RocksDBException | IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * Retrieve a value for a given key in a column family.
+     *
+     * @param columnFamilyName Column Family Name
+     * @param key Key to be retrieved
+     */
+    public V get(String columnFamilyName, K key) {
+        Preconditions.checkArgument(!closed);
+        try {
+            byte[] val = getRocksDB().get(managedHandlesMap.get(columnFamilyName), serializeKey(key));
+            return val == null ? null : deserializeValue(val);
+        } catch (RocksDBException | IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * Perform a prefix search and return stream of key-value pairs retrieved.
+     * @note This stream must be closed after use, otherwise it will cause a memory leak
+     *
+     * @param columnFamilyName Column Family Name
+     * @param prefix Prefix Key
+     */
+    public Stream<Tuple2<K, V>> prefixSearch(String columnFamilyName, byte[] prefix) {
+        Preconditions.checkArgument(!closed);
+        final RocksIterator it = getRocksDB().newIterator(managedHandlesMap.get(columnFamilyName));
+        Iterator<Tuple2<K, V>> conditionalIt = new ConditionalIteratorWrapper<>(it, this, prefix);
+        return StreamSupport
+                .stream(Spliterators.spliteratorUnknownSize(conditionalIt, 0), false)
+                .onClose(() -> it.close());
+    }
+
+    /**
+     * Perform a prefix delete and return stream of key-value pairs retrieved.
+     *
+     * @param columnFamilyName Column Family Name
+     * @param prefix Prefix Key
+     */
+    public void prefixDelete(String columnFamilyName, String prefix) {
```

Comment:
Maybe we shouldn't add code that is not used at present since it will add up the test cost